reframe/core/pipeline.py

```diff
@@ -1666,63 +1691,52 @@ def check_performance(self):
         '''
 
         with osext.change_dir(self._stagedir):
-            if hasattr(self, '_rfm_perf_vars') or self._rfm_perf_fns:
+            if self.perf_variables or self._rfm_perf_fns:
                 if hasattr(self, 'perf_patterns'):
                     raise ReframeSyntaxError(
                         f"assigning a value to 'perf_pattenrs' conflicts ",
-                        f"with using the 'performance_variables' decorator ",
+                        f"with using the 'performance_function' decorator ",
+                        f"or setting a value to 'perf_variables' "
                         f"(class {self.__class__.__qualname__})"
                     )
 
-                # Build the performance dict
-                if not hasattr(self, '_rfm_perf_vars'):
-                    self.perf_patterns = {}
-                    for fn in self._rfm_perf_fns:
-                        self.perf_patterns[fn._rfm_perf_name] = fn(self)
-
-                else:
-                    self.perf_patterns = self._rfm_perf_vars()
-
                 # Log the performance variables
                 self._setup_perf_logging()
-                for tag, expr in self.perf_patterns.items():
+                for tag, expr in self.perf_variables.items():
                     try:
-                        value, unit = sn.evaluate(expr)
-                        unit = getattr(unit, '_rfm_perf_units', None)
-                        if not unit:
-                            raise TypeError('Trigger the TypeError below')
-
-                    except TypeError:
-                        raise ReframeSyntaxError(
-                            f'function assigned for performance variable '
-                            f'{tag!r} is not a performance function'
+                        value = expr.evaluate()
+                        unit = expr.unit
+                    except Exception as e:
+                        logging.getlogger().warning(
+                            f'skipping evaluation of performance variable '
+                            f'{tag!r} in test {self.name!r}: {e}'
```

Comment:
Again, we don't need the test name here. The logger knows the context of the test, so it could log the test name regardless.